gustaf/spline/create.py

```diff
@@ -254,6 +255,289 @@ def revolved(
     return type(spline)(**spline_dict)
 
 
+def line(points, degree=1):
+    """Create a spline with the provided points as control points with given
+    degree.
+
+    Parameters
+    ----------
+    points : numpy.ndarray
+      npoints x ndims array of control points
+    degree : int, optional
+      Desired spline degree, by default 1
+
+    Returns
+    -------
+    gustaf.Bspline
+      Bspline with choosen control points and if necessary internal knots.
+    """
+    from gustaf import BSpline
+    cps = points
+    nknots = cps.shape[0] + degree + 1
+
+    knots = np.concatenate(
+            (
+                    np.full(degree,
+                            0.), np.linspace(0., 1., nknots - 2 * degree),
+                    np.full(degree, 1.0)
+            )
+    )
+
+    return BSpline(control_points=cps, knot_vectors=[knots], degrees=[degree])
+
+
+def circle(r):
+    """Line circle with radius r in the x-y plane around the origin.
+    The spline has an open knot vector and degree 2.
+
+    Parameters
+    ----------
+    r : float
+      radius
+
+    Returns
+    -------
+    gustaf.NURBS
+      Line circle NURBS
+    """
+
+    from gustaf.spline import NURBS
+
+    cps = np.array(
+            [
+                    [r, 0., 0.], [r, -r, 0.], [0., -r, 0.], [-r, -r, 0.],
+                    [-r, 0., 0.], [-r, r, 0.], [0., r, 0.], [r, r, 0.],
+                    [r, 0., 0.]
+            ]
+    )
+
+    weights = np.tile([1.0, 1 / np.sqrt(2)], 5)[:-1]
+    knots = np.repeat([0.0, 0.25, 0.5, 0.75, 1.0], [3, 2, 2, 2, 3])
+    degree = 2
+
+    return NURBS(
+            control_points=cps,
+            knot_vectors=[knots],
+            degrees=[degree],
+            weights=weights
+    )
+
+
+def rectangle(a, b):
+    """ Surface spline in rectangluar shape with linear degree.
+    The rectangle lies in the x-y plane with one corner in the origin.
+
+    Parameters
+    ----------
+    a : float
+      dimension of the rectangle in x-direction
+    b : float
+      dimension of the rectangle in y-direction
+    Returns
+    -------
+    gustaf.Bezier
+      Rectangluar Bezier spline with lengths of the sides a and b.
+    """
+    from gustaf.spline import Bezier
+
+    cps = np.zeros([4, 3])
+    cps[1, 0] = a
+    cps[2, 1] = b
+    cps[3, :2] = a, b
+
+    return Bezier(control_points=cps, degrees=[1, 1])
+
+
+def box(a, b, h):
+    """Volumetric spline with linear degree and dimensions a,b,h in
+    x,y,z-direction
+
+    Parameters
+    ----------
+    a : float
+      dimension of the box in x-direction
+    b : float
+      dimension of the box in y-direction
+    h : float
+      dimension of the box in z-direction
+
+    Returns
+    -------
+    gustaf.Bezier
+      Volumetric linear Bezier spline
+    """
+    return rectangle(a, b).create.extruded([0., 0., h])
+
+
+def disk(R, r0=0., angle=360., n_knot_spans=4):
```

Comment:
```suggestion
def disk(outer_radius, inner_radius=0., angle=360., n_knot_spans=4):
```
Try to be a bit more describtive with your arguments

Reply:
****